infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -91,61 +91,64 @@ public long query(CorfuStoreMetadata.Timestamp timestamp, LogReplicationMetadata
+        return query(null, LogReplicationMetadataKey.KeyType.SNAPSHOT_SEQ_NUM);
     }
 
     public long getLastProcessedLogTimestamp() {
-        return query(null, LogReplicationMetadataType.LAST_LOG_PROCESSED);
+        return query(null, LogReplicationMetadataKey.KeyType.LAST_LOG_ENTRY_PROCESSED);
     }
 
-    public void appendUpdate(TxBuilder txBuilder, LogReplicationMetadataType key, long val) {
-        LogReplicationMetadataKey txKey = LogReplicationMetadataKey.newBuilder().setKey(key.getVal()).build();
+    public void appendUpdate(TxBuilder txBuilder, LogReplicationMetadataKey.KeyType type, long val) {
+        LogReplicationMetadataKey txKey = LogReplicationMetadataKey.newBuilder().setKey(type).build();
         LogReplicationMetadataVal txVal = LogReplicationMetadataVal.newBuilder().setVal(Long.toString(val)).build();
         txBuilder.update(metadataTableName, txKey, txVal, null);
     }
 
-    private void appendUpdate(TxBuilder txBuilder, LogReplicationMetadataType key, String val) {
-        LogReplicationMetadataKey txKey = LogReplicationMetadataKey.newBuilder().setKey(key.getVal()).build();
+    private void appendUpdate(TxBuilder txBuilder, LogReplicationMetadataKey.KeyType key, String val) {
+        LogReplicationMetadataKey txKey = LogReplicationMetadataKey.newBuilder().setKey(key).build();
         LogReplicationMetadataVal txVal = LogReplicationMetadataVal.newBuilder().setVal(val).build();
         txBuilder.update(metadataTableName, txKey, txVal, null);
     }
 
     public void setupTopologyConfigId(long topologyConfigId) {
         CorfuStoreMetadata.Timestamp timestamp = corfuStore.getTimestamp();
-        long persistedTopologyConfigId = query(timestamp, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID);
+        long persistedTopologyConfigId = query(timestamp, LogReplicationMetadataKey.KeyType.TOPOLOGY_CONFIG_ID);
 
         if (topologyConfigId <= persistedTopologyConfigId) {
             log.warn("Skip setupTopologyConfigId. the current topologyConfigId " + topologyConfigId + " is not larger than the persistedTopologyConfigID " + persistedTopologyConfigId);
-            return;
         }
 
         TxBuilder txBuilder = corfuStore.tx(namespace);
 
-        for (LogReplicationMetadataType key : LogReplicationMetadataType.values()) {
+        // TODO pankti: Change to set the topology config id and not other fields.  This cannot be changed now as it
+        // fails later if anything is found not set.
+        for (LogReplicationMetadataKey.KeyType type : LogReplicationMetadataKey.KeyType.values()) {
             long val = Address.NON_ADDRESS;
-            if (key == LogReplicationMetadataType.TOPOLOGY_CONFIG_ID) {
+            if (type == LogReplicationMetadataKey.KeyType.TOPOLOGY_CONFIG_ID) {
                 val = topologyConfigId;
             }
-            appendUpdate(txBuilder, key, val);
+            if (type != LogReplicationMetadataKey.KeyType.UNRECOGNIZED) {
```

Comment:
I'm curious, if we update the topologyConfigId, do we always need to reset all other values? Let's say a new standby is added.. is this still needed?

Reply:
I think we should not update other values in this function, because it causes more TX conflicts, and discovery service will shutdown. If the version filed is empty, we should initialize version somewhere...

Reply:
If the active cluster doesn't change, the source of data doesn't change, we don't need to reset the log replication metadata. So, we should have two API s:
1. just  updateTopologyConfigId, if the active cluster doesn't change. 
2. update Topology Id and reset logReplicationMetadata if the active cluster change, as the source of data change.